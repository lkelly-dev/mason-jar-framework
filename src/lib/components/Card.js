import React, { useContext } from 'react';
import { cx } from 'emotion';
import { debugAttrs, debugMode } from '../utils/debugging';

const Card = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardClass = cx('card', className);
  return (
    <div className={cardClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const CardTitle = props => {
  const { context, children, className, as, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardTitleClass = cx('card-title', 'mb-0', className);
  const Component = as || 'h2';
  return (
    <Component className={cardTitleClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </Component>
  );
};

const CardHeader = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const headerClass = cx('card-header', className);
  return (
    <div className={headerClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const CardBody = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardBodyClass = cx('card-body', className);
  return (
    <div className={cardBodyClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const CardText = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardTextClass = cx('card-text', className);
  return (
    <span className={cardTextClass} 
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </span>
  );
};

const CardDropdown = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardDropdownClass = cx('dropdown', 'card-dropdown', className);
  return (
    <div className={cardDropdownClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const CardAvatar = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardAvatarClass = cx('card-avatar', 'avatar', className);
  return (
    <div className={cardAvatarClass} 
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const CardTable = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  const cardTableClass = cx('card-table', className);
  return (
    <table className={cardTableClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </table>
  );
};

Card.Header = CardHeader;
Card.Body = CardBody;
Card.Text = CardText;
Card.Title = CardTitle;
Card.Dropdown = CardDropdown;
Card.Avatar = CardAvatar;
Card.Table = CardTable;

export { Card, CardHeader, CardBody, CardText, CardTitle, CardDropdown, CardAvatar, CardTable };
