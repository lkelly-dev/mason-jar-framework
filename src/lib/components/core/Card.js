import React from 'react';
import { cx } from 'emotion';

const Card = props => {
  const { children, className, ...rest } = props;
  const cardClass = cx(className, 'card');
  return (
    <div className={cardClass} {...rest}>
      {children}
    </div>
  );
};

const CardTitle = props => {
  const { children, className, as, ...rest } = props;
  const cardTitleClass = cx(className, 'card-title', 'mb-0');
  const Component = as || 'h2';
  return (
    <Component className={cardTitleClass} {...rest}>
      {children}
    </Component>
  );
};

const CardHeader = props => {
  const { children, className, ...rest } = props;
  const headerClass = cx(className, 'card-header');
  return (
    <div className={headerClass} {...rest}>
      {children}
    </div>
  );
};

const CardBody = props => {
  const { children, className, ...rest } = props;
  const cardBodyClass = cx(className, 'card-body');
  return (
    <div className={cardBodyClass} {...rest}>
      {children}
    </div>
  );
};

const CardText = props => {
  const { children, className, ...rest } = props;
  const cardTextClass = cx(className, 'card-text');
  return (
    <span className={cardTextClass} {...rest}>
      {children}
    </span>
  );
};

const CardDropdown = props => {
  const { children, className, ...rest } = props;
  const cardDropdownClass = cx(className, 'dropdown', 'card-dropdown');
  return (
    <div className={cardDropdownClass} {...rest}>
      {children}
    </div>
  );
};

const CardAvatar = props => {
  const { children, className, ...rest } = props;
  const cardAvatarClass = cx(className, 'card-avatar', 'avatar');
  return (
    <div className={cardAvatarClass} {...rest}>
      {children}
    </div>
  );
};

// TODO: Test this
const CardTable = props => {
  const { children, className, ...rest } = props;
  const cardTableClass = cx(className, 'card-table');
  return (
    <table className={cardTableClass} {...rest}>
      {children}
    </table>
  );
};

Card.Header = CardHeader;
Card.Body = CardBody;
Card.Text = CardText;
Card.Title = CardTitle;
Card.Dropdown = CardDropdown;
Card.Avatar = CardAvatar;
Card.Table = CardTable;

export { Card, CardHeader, CardBody, CardText, CardTitle, CardDropdown, CardAvatar, CardTable };
