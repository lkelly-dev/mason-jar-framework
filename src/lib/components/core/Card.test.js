import React from 'react';
import { shallow } from 'enzyme';
import { Card, CardHeader, CardBody, CardText, CardTitle, CardDropdown, CardAvatar, CardTable } from "./Card";

describe('Card', () => {
  it('Card matches snapshot', () => {
    const component = shallow(<Card></Card>);
    expect(component).toMatchSnapshot();
  })
});
