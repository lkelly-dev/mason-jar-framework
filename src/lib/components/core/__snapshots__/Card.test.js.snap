// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card matches snapshot 1`] = `
<div
  className="card"
/>
`;
